nexus/db-queries/src/db/pub_test_utils/mod.rs

```diff
@@ -0,0 +1,316 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Test support code that can be enabled by dependencies via this crate's
+//! `testing` feature.
+//!
+//! This feature should only be enabled under `dev-dependencies` to avoid this
+//! test support code leaking into release binaries.
+
+use crate::authz;
+use crate::context::OpContext;
+use crate::db;
+use crate::db::DataStore;
+use omicron_test_utils::dev::db::CockroachInstance;
+use slog::Logger;
+use std::sync::Arc;
+use uuid::Uuid;
+
+pub mod crdb;
+
+enum Populate {
+    Nothing,
+    Schema,
+    SchemaAndData,
+}
+
+enum Interface {
+    Nothing,
+    Pool,
+    Datastore,
+}
+
+fn new_pool(log: &Logger, db: &CockroachInstance) -> Arc<db::Pool> {
+    let cfg = db::Config { url: db.pg_config().clone() };
+    Arc::new(db::Pool::new_single_host(log, &cfg))
+}
+
+pub struct TestDatabaseBuilder {
+    populate: Populate,
+    interface: Interface,
+}
+
+impl TestDatabaseBuilder {
+    /// Creates a new database buidler.
+    pub fn new() -> Self {
+        Self { populate: Populate::Nothing, interface: Interface::Nothing }
+    }
+
+    /// Populates the database without a schema
+    pub fn without_schema(self) -> Self {
+        self.populate(Populate::Nothing)
+    }
+
+    /// Populates the database with a schema
+    pub fn with_schema(self) -> Self {
+        self.populate(Populate::Schema)
+    }
+
+    /// Populates the database with a schema and loads it with data
+    pub fn with_schema_and_builtin_data(self) -> Self {
+        self.populate(Populate::SchemaAndData)
+    }
```

Comment:
It might be nice if the API more clearly indicated that these options were mutually exclusive. Perhaps making the `Populate` enum public and just having one method that sets it, or naming them `populate_...` or something?

Reply:
Agreed, updated to `populate_`